src/org/mozilla/javascript/NativeConsole.java

```diff
@@ -325,7 +325,34 @@ private static String formatObj(Context cx, Scriptable scope, Object arg) {
         }
 
         try {
-            Object stringify = NativeJSON.stringify(cx, scope, arg, null, null);
+            // NativeJSON.stringify outputs Callable's as null, convert to string
+            // to make the output less confusing
+            final Callable replacer =
+                    new Callable() {
+                        @Override
+                        public Object call(
+                                Context callCx,
+                                Scriptable callScope,
+                                Scriptable callThisObj,
+                                Object[] callArgs) {
+                            Object value = callArgs[1];
+                            while (value instanceof Delegator) {
+                                value = ((Delegator) value).getDelegee();
+                            }
+                            if (value instanceof BaseFunction) {
+                                StringBuilder sb = new StringBuilder();
+                                sb.append("function ")
+                                        .append(((BaseFunction) value).getFunctionName())
+                                        .append("() {}");
```

Comment:
Wondering whether this should not be `"() {...}"` to make it clearer that the body content was stripped